From bbe8b4b6f963e2fd5a2ea885d14b827448e3003c Mon Sep 17 00:00:00 2001
From: Cecylia Bocovich <cohosh@torproject.org>
Date: Wed, 4 Sep 2019 18:03:37 +0000
Subject: [PATCH] Allow gathering of candidates to generate offer

A change was recently made to the SetLocal and SetRemoteDescription
functions for PeerConnection that starts ICE gathering for the trickle
method only when generating an answer. This means clients generating an
offer to initiate signaling will stall indefinitely.
---
 peerconnection.go | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/peerconnection.go b/peerconnection.go
index 8e64b4b..16c39c6 100644
--- a/peerconnection.go
+++ b/peerconnection.go
@@ -832,10 +832,7 @@ func (pc *PeerConnection) SetLocalDescription(desc SessionDescription) error {
 		return nil
 	}
 
-	if desc.Type == SDPTypeAnswer {
-		return pc.iceGatherer.Gather()
-	}
-	return nil
+	return pc.iceGatherer.Gather()
 }
 
 // LocalDescription returns pendingLocalDescription if it is not null and
@@ -1027,9 +1024,6 @@ func (pc *PeerConnection) SetRemoteDescription(desc SessionDescription) error {
 		pc.mu.Unlock()
 	}()
 
-	if (desc.Type == SDPTypeAnswer || desc.Type == SDPTypePranswer) && pc.iceGatherer.agentIsTrickle {
-		return pc.iceGatherer.Gather()
-	}
 	return nil
 }
 
-- 
2.20.1

